### PR TITLE
[RHOAIENG-4334] 'Job' Name (in code) alignment

### PR DESCRIPTION
**Pull request by dpanshug**
Closes: [RHOAIENG-4334](https://issues.redhat.com/browse/RHOAIENG-4334)

## Description
Renamed "Job" to "Recurring run"

## How Has This Been Tested?
Check the codebase to ensure that all instances of 'job' are either replaced with 'recurring run' or that the corresponding function/type is marked as deprecated. 

Also, run and check all the functionality of pipelines in dashboard to ensure everything is working as expected.

## Test Impact
No tests added, modified the previous tests to accommodate the change.

## Request review criteria:

Self checklist (all need to be checked):
- [x] The developer has manually tested the changes and verified that the changes work
- [x] Commits have been squashed into descriptive, self-contained units of work (e.g. 'WIP' and 'Implements feedback' style messages have been removed)
- [x] Testing instructions have been added in the PR body (for PRs involving changes that are not immediately obvious).
- [x] The developer has added tests or explained why testing cannot be added (unit or cypress tests for related changes)

If you have UI changes: 

- [ ] Included any necessary screenshots or gifs if it was a UI change.
- [ ] Included tags to the UX team if it was a UI/UX change (find relevant UX in the [SMEs](https://github.com/opendatahub-io/odh-dashboard/tree/main/docs/smes.md) section).

After the PR is posted & before it merges:
- [ ] The developer has tested their solution on a cluster by using the image produced by the PR to `main`

### CODE REVIEW

File: frontend/src/api/pipelines/custom.ts [hunk 1]
```diff
@@ -5,19 +5,19 @@ import {
   GetPipelineAPI,
   DeletePipelineAPI,
   ListPipelinesRunAPI,
-  ListPipelinesRunJobAPI,
+  ListPipelineRecurringRunsAPI,
   ListPipelinesAPI,
   UploadPipelineAPI,
-  UpdatePipelineRunJobAPI,
+  UpdatePipelineRecurringRunsAPI,
```
Comment (ppadti) [inline frontend/src/api/pipelines/custom.ts:11]:
```suggestion
  UpdatePipelineRecurringRunAPI,
```

File: frontend/src/api/pipelines/custom.ts [hunk 2]
```diff
@@ -5,19 +5,19 @@ import {
   GetPipelineAPI,
   DeletePipelineAPI,
   ListPipelinesRunAPI,
-  ListPipelinesRunJobAPI,
+  ListPipelineRecurringRunsAPI,
```
Comment (ppadti) [inline frontend/src/api/pipelines/custom.ts:8]: I am not sure whether is has to be `ListPipelinesRecurringRunAPI` or is it fine to use `ListPipelineRecurringRunsAPI` here.

File: frontend/src/concepts/pipelines/types.ts [hunk 3]
```diff
@@ -103,19 +106,19 @@ export type ListPipelineRuns = (
...
-) => Promise<ListPipelineRunJobsResourceKF>;
+) => Promise<ListPipelineRecurringRunsResourceKF>;
 export type ListPipelineVersions = (
   opts: K8sAPIOptions,
   pipelineId: string,
   params?: PipelineParams,
 ) => Promise<ListPipelineVersionsKF>;
 export type UpdatePipelineRun = (opts: K8sAPIOptions, runId: string) => Promise<void>;
-export type UpdatePipelineRunJob = (
+export type UpdatePipelineRecurringRuns = (
```
Comment (ppadti) [inline frontend/src/concepts/pipelines/types.ts:119]:
```suggestion
export type UpdatePipelineRecurringRun = (
```

**Review by ppadti — CHANGES_REQUESTED** [review]

File: frontend/src/__tests__/cypress/cypress/tests/mocked/pipelines/pipelineRuns.cy.ts [hunk 4]
```diff
@@ -775,7 +775,7 @@ describe('Pipeline runs', () => {
             path: { namespace: projectName, serviceName: 'dspa' },
           },
           {
-            recurringRuns: mockJobRuns.slice(0, 10),
+            recurringRuns: mockRuns.slice(0, 10),
```
[the remark this answers is not in the document]
Comment (dpanshug) [reply frontend/src/__tests__/cypress/cypress/tests/mocked/pipelines/pipelineRuns.cy.ts:778]: there's already a variable named mockRecurringRuns on line 98, so can't use here.

File: frontend/src/__tests__/cypress/cypress/tests/mocked/pipelines/pipelineRuns.cy.ts [hunk 5]
```diff
@@ -805,7 +805,7 @@ describe('Pipeline runs', () => {
             path: { namespace: projectName, serviceName: 'dspa' },
           },
           {
-            recurringRuns: mockJobRuns.slice(10, 15),
+            recurringRuns: mockRuns.slice(10, 15),
```
[the remark this answers is not in the document]
Comment (dpanshug) [reply frontend/src/__tests__/cypress/cypress/tests/mocked/pipelines/pipelineRuns.cy.ts:808]: there's already a variable named mockRecurringRuns on line 98, so can't use here.

File: frontend/src/__tests__/cypress/cypress/tests/mocked/pipelines/pipelineRuns.cy.ts [hunk 6]
```diff
@@ -834,15 +834,15 @@ describe('Pipeline runs', () => {
             path: { namespace: projectName, serviceName: 'dspa' },
           },
           {
-            recurringRuns: mockJobRuns.slice(0, 10),
+            recurringRuns: mockRuns.slice(0, 10),
```
[the remark this answers is not in the document]
Comment (dpanshug) [reply frontend/src/__tests__/cypress/cypress/tests/mocked/pipelines/pipelineRuns.cy.ts:837]: same reason mentioned above

File: frontend/src/api/pipelines/custom.ts (hunk 2 above)
Comment (dpanshug) [reply frontend/src/api/pipelines/custom.ts:8]: i think this one is fine

**Review by Gkrumbach07 — COMMENTED** [review]
/lgtm

going to test live to make sure things still work

**Review by manaswinidas — APPROVED** [review]
/lgtm

**Comment by Gkrumbach07** [comment]
tested job flows and things work as expected

/lgtm
/approve

**Pull request merged**